spring/spring6/src/main/java/com/linecorp/armeria/spring/client/ArmeriaHttpExchangeAdapter.java

```diff
@@ -0,0 +1,245 @@
+import org.springframework.web.service.invoker.HttpRequestValues;
+import org.springframework.web.service.invoker.ReactiveHttpRequestValues;
+import org.springframework.web.service.invoker.ReactorHttpExchangeAdapter;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+import org.springframework.web.util.UriBuilderFactory;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SplitHttpResponse;
+import com.linecorp.armeria.spring.internal.client.ArmeriaClientHttpRequest;
+import com.linecorp.armeria.spring.internal.client.ArmeriaClientHttpResponse;
+import com.linecorp.armeria.spring.internal.common.DataBufferFactoryWrapper;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * A {@link ReactorHttpExchangeAdapter} implementation for the Armeria {@link WebClient}.
+ *
+ * <p><pre>{@code
+ * import com.linecorp.armeria.client.WebClient;
+ * import org.springframework.web.service.invoker.HttpServiceProxyFactory;
+ *
+ * WebClient webClient = ...;
+ * ArmeriaHttpExchangeAdapter adapter = ArmeriaHttpExchangeAdapter.of(webClient);
+ * MyService service =
+ *   HttpServiceProxyFactory.builderFor(adapter)
+ *                          .build()
+ *                          .createClient(MyService.class);
+ * }</pre>
+ */
+public final class ArmeriaHttpExchangeAdapter extends AbstractReactorHttpExchangeAdapter {
+
+    /**
+     * Creates a new instance with the specified {@link WebClient}.
+     */
+    public static ArmeriaHttpExchangeAdapter of(WebClient webClient) {
+        return of(webClient, ExchangeStrategies.withDefaults());
+    }
+
+    /**
+     * Creates a new instance with the specified {@link WebClient} and {@link ExchangeStrategies}.
+     */
+    public static ArmeriaHttpExchangeAdapter of(WebClient webClient, ExchangeStrategies exchangeStrategies) {
+        requireNonNull(webClient, "webClient");
+        requireNonNull(exchangeStrategies, "exchangeStrategies");
+        return new ArmeriaHttpExchangeAdapter(webClient, exchangeStrategies);
+    }
+
+    private final WebClient webClient;
+    private final UriBuilderFactory uriBuilderFactory;
+    private final ExchangeStrategies exchangeStrategies;
+
+    private ArmeriaHttpExchangeAdapter(WebClient webClient, ExchangeStrategies exchangeStrategies) {
+        this.webClient = webClient;
+        this.exchangeStrategies = exchangeStrategies;
+        final URI baseUri = webClient.uri();
+        if (Clients.isUndefinedUri(baseUri)) {
+            uriBuilderFactory = new DefaultUriBuilderFactory();
+        } else {
+            uriBuilderFactory = new DefaultUriBuilderFactory(UriComponentsBuilder.fromUri(baseUri));
```

Comment:
Good point. I realized that this logic double-prefixes a client's path to request paths. Because `WebClient` internally added the prefix again. I removed `DefaultUriBuilderFactory` created from `baseUri`.